### PR TITLE
Fix the styles for StyleCop

### PR DESCRIPTION
**Pull request by mattleibow**
These are the changes required to make StyleCop happy. This should be merged in before #34 otherwise the build will fail.

IMHO: I like harsh rules because they force me into the patter of the code. I think we should go harsh and then, if this causes any problems, slowly relax them. But they aren't to harsh, we shouldn't leave blank lines all over the place - or forget to space out our members.

### CODE REVIEW

File: Caboodle.Tests/Geocoding_Tests.cs [hunk 1]
```diff
@@ -1,8 +1,5 @@
-﻿using Microsoft.Caboodle;
-using System;
-using System.Collections.Generic;
-using System.Text;
-using System.Threading.Tasks;
+﻿using System.Threading.Tasks;
+using Microsoft.Caboodle;
```
Comment (mattleibow) [inline Caboodle.Tests/Geocoding_Tests.cs]: Namespaces rules:

 - System.* first
 - Alphabetically

File: Caboodle.Tests/Preferences_Tests.cs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-using Xunit;
+﻿using Xunit;
```
Comment (mattleibow) [inline Caboodle.Tests/Preferences_Tests.cs]: Files must be UTF-8 with BOM

File: Caboodle/Geocoding/Geocoding.ios.cs [hunk 3]
```diff
@@ -19,7 +18,6 @@ public partial class Geocoding
 
         public static async Task<IEnumerable<Location>> GetLocationsAsync(string address)
         {
-
```
Comment (mattleibow) [inline Caboodle/Geocoding/Geocoding.ios.cs]: No blank lines after an opening brace.

File: Caboodle/Geocoding/Geocoding.ios.cs [hunk 4]
```diff
@@ -29,4 +27,3 @@ public partial class Geocoding
         }
     }
 }
-
```
Comment (mattleibow) [inline Caboodle/Geocoding/Geocoding.ios.cs]: A single blank line at the end of the file.

File: Caboodle/Geocoding/Geocoding.netstandard.cs [hunk 5]
```diff
@@ -8,7 +8,6 @@ public partial class Geocoding
         public static Task<IEnumerable<Placemark>> GetPlacemarksAsync(double latitude, double longitude) =>
             throw new NotImplentedInReferenceAssembly();
 
-
```
Comment (mattleibow) [inline Caboodle/Geocoding/Geocoding.netstandard.cs]: No chunks of blank lines - just a single line.

File: Caboodle/Geocoding/Geocoding.android.cs [hunk 6]
```diff
@@ -31,5 +30,4 @@ public partial class Geocoding
             }
         }
     }
-
 }
```
Comment (mattleibow) [inline Caboodle/Geocoding/Geocoding.android.cs]: No blank lines before a closing brace.

File: Caboodle/Platform/Platform.android.cs [hunk 7]
```diff
@@ -70,4 +70,4 @@ public void OnActivityStopped(Activity activity)
         {
         }
     }
-}
\ No newline at end of file
+}
```
Comment (mattleibow) [inline Caboodle/Platform/Platform.android.cs]: All file must end with a blank line.

File: Caboodle/Preferences/Preferences.shared.cs [hunk 8]
```diff
@@ -15,27 +15,37 @@ public Preferences(string sharedName)
 
         public string Get(string key, string defaultValue) =>
             Get<string>(key, defaultValue);
+
```
Comment (mattleibow) [inline Caboodle/Preferences/Preferences.shared.cs]: Methods and properties must have a blank line in between them.

File: Caboodle/Types/Location.shared.cs [hunk 9]
```diff
@@ -49,13 +48,15 @@ public Location(Location point)
...
             CalculateDistance(locationStart.Latitude, locationStart.Longitude, locationEnd.Latitude, locationEnd.Longitude, units);
 
         public static double CalculateDistance(
-            double latitudeStart, double latitudeEnd,
-            double longitudeStart, double longitudeEnd, DistanceUnits units)
+            double latitudeStart,
+            double latitudeEnd,
+            double longitudeStart,
+            double longitudeEnd,
+            DistanceUnits units)
```
Comment (mattleibow) [inline Caboodle/Types/Location.shared.cs]: All parameters must be on a single line, or all on separate lines.

File: Caboodle/Types/Location.shared.cs [hunk 10]
```diff
@@ -64,7 +65,7 @@ public Location(Location point)
             var rlat2 = PI * latitudeEnd / 180.0;
             var theta = longitudeStart - longitudeEnd;
             var rtheta = PI * theta / 180.0;
-            var dist = Sin(rlat1) * Sin(rlat2) + Cos(rlat1) * Cos(rlat2) * Cos(rtheta);
+            var dist = (Sin(rlat1) * Sin(rlat2)) + (Cos(rlat1) * Cos(rlat2) * Cos(rtheta));
```
Comment (mattleibow) [inline Caboodle/Types/Location.shared.cs]: Parenthesis must be use when precedence might be ambiguous.

File: Caboodle/Types/LocationExtensions.android.cs [hunk 11]
```diff
@@ -16,17 +16,17 @@ public static partial class LocationExtensions
...
         private static readonly DateTime epoch = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
+
         internal static DateTimeOffset GetTimestamp(this Android.Locations.Location location)
         {
             try
             {
                 return new DateTimeOffset(epoch.AddMilliseconds(location.Time));
             }
-            catch (Exception e)
+            catch (Exception)
```
Comment (mattleibow) [inline Caboodle/Types/LocationExtensions.android.cs]: Unused variable. Warnings as errors.

File: DeviceTests/Caboodle.DeviceTests.Android/Properties/AssemblyInfo.cs [hunk 12]
```diff
@@ -3,7 +3,7 @@
 using System.Runtime.InteropServices;
 using Android.App;
 
-// General Information about an assembly is controlled through the following
```
Comment (mattleibow) [inline DeviceTests/Caboodle.DeviceTests.Android/Properties/AssemblyInfo.cs]: Comments can't have trailing whitespace.

File: Samples/Caboodle.Samples/ViewModel/GeocodingViewModel.cs [hunk 13]
```diff
@@ -13,8 +11,9 @@ public class GeocodingViewModel : BaseViewModel
     {
         public GeocodingViewModel()
         {
-            //Set UWP Map Key
+            // set UWP Map Key
```
Comment (mattleibow) [inline Samples/Caboodle.Samples/ViewModel/GeocodingViewModel.cs]: Comments must start with a space to separate from the `//`

File: DeviceTests/Caboodle.DeviceTests.Android/MainActivity.cs [hunk 14]
```diff
@@ -14,9 +14,9 @@ protected override void OnCreate(Bundle bundle)
             // tests can be inside the main assembly
             AddTestAssembly(Assembly.GetExecutingAssembly());
             AddExecutionAssembly(typeof(MainActivity).Assembly);
-            // or in any reference assemblies
```
Comment (mattleibow) [inline DeviceTests/Caboodle.DeviceTests.Android/MainActivity.cs]: Single line comments `//` must begin with a blank line.

**Comment by jamesmontemagno** [comment]
Amazing

**Pull request merged**